src/Process/PointCloudProcessing.js

```diff
@@ -163,18 +163,49 @@ function markForDeletion(elt) {
 }
 
 export default {
-    preUpdate(context, layer) {
-        // TODO: use changeSource
-        layer.counters = {
-            pointCount: 0,
-            displayedCount: 0,
-        };
-
+    preUpdate(context, layer, changeSources) {
         // Bail-out if not ready
         if (!layer.root) {
             return [];
         }
 
+        if (changeSources.has(undefined) || changeSources.size == 0) {
+            return [layer.root];
+        }
+
+        // lookup lowest common ancestor of changeSources
+        let commonAncestorName;
+        for (const source of changeSources.values()) {
+            if (source.isCamera) {
+                // if the change is caused by a camera move, no need to bother
+                // to find common ancestor: we need to update the whole tree:
+                // some invisible tiles may now be visible
+                return [layer.root];
+            }
+            if (source.obj === undefined) {
+                continue;
+            }
+            if (source.obj.isPoints && source.obj.layer == layer.id) {
+                if (!commonAncestorName) {
+                    commonAncestorName = source.name;
+                } else {
+                    let i;
+                    for (i = 0; i < Math.min(source.name.length, commonAncestorName.length); i++) {
+                        if (source.name[i] != commonAncestorName[i]) {
+                            break;
+                        }
+                    }
+                    commonAncestorName = commonAncestorName.substr(0, i);
+                    if (commonAncestorName.length == 0) {
+                        break;
+                    }
+                }
+            }
+        }
+        if (commonAncestorName && commonAncestorName.length > 0) {
```

Comment:
(neat) I guess the second test (after `&&`) is useless, so this could be simplified to `if (commonAncestorName)`.

Reply:
Indeed: fixed.